api/src/main/java/org/openmrs/module/commonreports/reports/AntenatalGestationReportManager.java

```diff
@@ -0,0 +1,216 @@
+public class AntenatalGestationReportManager extends ActivatedReportManager {
+	
+	@Autowired
+	private InitializerService inizService;
+	
+	@Override
+	public boolean isActivated() {
+		return inizService.getBooleanFromKey("report.antenatalGestation.active", false);
+	}
+	
+	@Override
+	public String getVersion() {
+		return "1.0.0-SNAPSHOT";
+	}
+	
+	@Override
+	public String getUuid() {
+		return "b6b2a422-38d1-4fb1-acfc-46eb842bcf7f";
+	}
+	
+	@Override
+	public String getName() {
+		return MessageUtil.translate("commonreports.report.antenatalGestation.reportName");
+	}
+	
+	@Override
+	public String getDescription() {
+		return MessageUtil.translate("commonreports.report.antenatalGestation.reportDescription");
+	}
+	
+	private Parameter getStartDateParameter() {
+		return new Parameter("startDate", "Start Date", Date.class);
+	}
+	
+	private Parameter getEndDateParameter() {
+		return new Parameter("endDate", "End Date", Date.class);
+	}
+	
+	public static String col1 = "";
+	
+	public static String col2 = "";
+	
+	public static String col3 = "";
+	
+	public static String col4 = "";
+	
+	public static String col5 = "";
+	
+	public static String col6 = "";
+	
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> params = new ArrayList<Parameter>();
+		params.add(getStartDateParameter());
+		params.add(getEndDateParameter());
+		return params;
+	}
+	
+	@Override
+	public ReportDefinition constructReportDefinition() {
+		
+		ReportDefinition rd = new ReportDefinition();
+		rd.setUuid(getUuid());
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		
+		rd.setParameters(getParameters());
+		
+		CohortCrossTabDataSetDefinition antenatalGestation = new CohortCrossTabDataSetDefinition();
+		antenatalGestation.addParameters(getParameters());
+		rd.addDataSetDefinition(getName(), Mapped.mapStraightThrough(antenatalGestation));
+		
+		Map<String, Object> parameterMappings = new HashMap<String, Object>();
+		parameterMappings.put("onOrAfter", "${startDate}");
+		parameterMappings.put("onOrBefore", "${endDate}");
+		
+		ConceptService cs = Context.getConceptService();
+		String[] myArray2 = { "0-13", "14-27", "28-40", "Total" };
+		
+		for (String member : myArray2) {
+			
+			if (member.equals("Total")) {
+				
+				SqlCohortDefinition sqd = new SqlCohortDefinition();
+				String st = "select person_id from obs where concept_id = (select DISTINCT concept_id from concept_name where name = 'Number of Weeks' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0)  AND obs_group_id IN (select obs_id from obs where concept_id = (select DISTINCT concept_id from concept_name where name = 'Estimated Gestational Age' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0))";
+				sqd.setQuery(st);
+				sqd.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+				sqd.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
```

Comment:
I'd use a `CodedObsCohortDefinition` instead and provide the concepts through configuration.